src/pages/JoinPage.tsx

```diff
@@ -18,18 +19,30 @@ export default function JoinPage() {
   const navigate = useNavigate();
   const dispatch = useAppDispatch();
 
-  const { hasJoinedGame} = useAppSelector(state => state.game);
+  const { hasJoinedGame } = useAppSelector(state => state.game);
+
+  // Clear all ccookies if we are joining as a player. This is neccessary
+  // to not fuck with the state
+  useClearCookies();
```

Comment:
Having `useClearCookies` in the JoinPage actually creates some issues so I removed it in my PR. We only need to clear the cookies on game end